src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesigntimeLicenseContext.cs

```diff
@@ -103,14 +100,11 @@ public override string GetSavedLicenseKey(Type type, Assembly resourceAssembly)
                         }
                     }
                 }
-                else if (!resourceAssembly.IsDynamic)
+                // Location is not supported by emitted assemblies and this will throw an exception
+                // down below.
+                else if (!resourceAssembly.IsDynamic && resourceAssembly.Location != string.Empty)
```

Comment:
```suggestion
                else if (!resourceAssembly.IsDynamic)
```

I do not think it is necessary to check for empty location up-front here.

Reply:
@jkotas do you think it should be something more like

```csharp
else if (!resourceAssembly.IsDynamic)
{
    // Emitted ssemblies return empty string from Location.
    var location = asm.Location;
    if (location != string.Empty) 
    {
        string fileName = new FileInfo(location).Name;
        ...
    }
}
```

Reply:
I think it can be:
```csharp
else
{
    string location = asm.Location;
    if (location != string.Empty)
    {
        string fileName = Path.GetFileName(location);
...
```

Reply:
Have updated as suggested, thanks!